src/Persistence/PersistenceManager.php

```diff
@@ -140,14 +143,53 @@ public function flush(ObjectManager $om): void
         }
     }
 
+    /**
+     * @template T of object
+     *
+     * @param T $object
+     */
+    public function autorefresh(object $object, object $clone): void
+    {
+        $strategy = $this->strategyFor($object::class);
+        $om = $strategy->objectManagerFor($object::class);
+
+        $id = $strategy->getIdentifierValues($clone);
+
+        if ($id) {
+            try {
+                $om->refresh($object);
+
+                if ($strategy->getIdentifierValues($object)) {
+                    // no identifier values means the object no longer exists
+                    return;
+                }
+            } catch (\Throwable $e) {
+            }
+
+            // let's detach the object, in order to prevent Doctrine cache
+            $om->detach($object);
+            if ($refreshedObject = $om->find($object::class, $id)) {
+                Hydrator::hydrateFromOtherObject($object, $refreshedObject);
+
+                return;
+            }
+        }
+
+        // the object no longer exists
+        Hydrator::hydrateFromOtherObject($object, $clone);
+        $om->detach($object);
```

Comment:
if the object no longer exist, we still need to return something, otherwise, the lazy ghost will be re-initialized as an empty shell. For this purpose we're using a clone of the object